project-base/storefront/hooks/cart/useAddOrderItemsToCart.ts

```diff
@@ -0,0 +1,37 @@
+import { useCurrentCart } from 'connectors/cart/Cart';
+import { useAddOrderItemsToCartMutationApi } from 'graphql/generated';
+import { getInternationalizedStaticUrls } from 'helpers/localization/getInternationalizedStaticUrls';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import { useRouter } from 'next/router';
+import { Dispatch, SetStateAction, useState } from 'react';
+
+export const useAddOrderItemsToCart = (): [
+    string | undefined,
+    Dispatch<SetStateAction<string | undefined>>,
+    (orderUuid: string) => void,
+    (orderUuid: string, shouldMerge: boolean) => void,
+] => {
+    const [orderUuid, setOrderUuid] = useState<string>();
+    const { cart, isCartEmpty } = useCurrentCart();
+    const router = useRouter();
+    const { url } = useDomainConfig();
+    const [cartUrl] = getInternationalizedStaticUrls(['/cart'], url);
+    const [, AddOrderItemsToCart] = useAddOrderItemsToCartMutationApi();
+
+    const addOrderItemsToEmptyCart = (orderUuid: string) => {
+        if (isCartEmpty) {
+            AddOrderItemsToCart({ input: { orderUuid, cartUuid: null, shouldMerge: false } });
+            router.push(cartUrl);
+        } else {
+            setOrderUuid(orderUuid);
+        }
+    };
+
+    const mergeOrderItemsWithCurrentCart = (orderUuid: string, shouldMerge: boolean) => {
```

Comment:
Also you can have default state for this `shouldMerge` 

```suggestion
    const mergeOrderItemsWithCurrentCart = (orderUuid: string, shouldMerge = false) => {
```

and pass `true` only when you need merge, otherwise it is not necessary pass any value

Reply:
Nice, done